server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -118,14 +132,23 @@ protected void doGet(HttpServletRequest req, HttpServletResponse res)
             return;
         }
 
-        String youtubeResBody = getYoutubeRes(API_KEY, accessToken.toString());
-        int numVideos = Integer.parseInt(req.getParameter("num_videos"));
+        String numVideosParam = req.getParameter("num_videos");
+        int numVideos = Integer.parseInt(DEFAULT_NUM_VIDS);
+        if (numVideosParam != null) {
+            numVideos = Integer.parseInt(numVideosParam);
+        }
+
+        String youtubeResBody = getYoutubeRes(API_KEY, accessToken.toString(), numVideos);
+        JsonObject youtubeJsonObj = JsonParser.parseString(youtubeResBody).getAsJsonObject();
 
         var genreCount = new HashMap<String, Integer>();
-        updateMusicCount(youtubeResBody, genreCount, numVideos);
+        updateMusicCount(youtubeJsonObj, genreCount);
+        int totalLiked = getTotalResults(youtubeJsonObj);
+        genreCount.put("totalLiked", totalLiked);
```

Comment:
"totalLiked" doesn't seem like it belongs in genreCount, since it's not a genre. Your result should hold that value separately from the genreCount map.

Reply:
I do that in my most recent PR! https://github.com/googleinterns/step99-2020/pull/46

Reply:
implemented your comments by creating custom java class that I sent to frontend, see new commits on this branch